### PR TITLE
expression: implement vectorized evaluation for `builtinCharLengthSig`

### PR DESCRIPTION
**Pull request by zhongshanhao**
### What problem does this PR solve? 
expression: implement vectorized evaluation for `builtinCharLengthSig` for #12106

### What is changed and how it works?
```
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinStringFunc/builtinCharLengthSig-VecBuiltinFunc-4         	  336915	      3565 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinStringFunc/builtinCharLengthSig-NonVecBuiltinFunc-4      	   29977	     38838 ns/op	       0 B/op	       0 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_string_vec.go [hunk 1]
```diff
@@ -1699,9 +1699,29 @@ func (b *builtinFromBase64Sig) vecEvalString(input *chunk.Chunk, result *chunk.C
...
+
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		str := buf.GetBytes(i)
+		i64s[i] = int64(len(str))
```
Comment (qw4990) [inline expression/builtin_string_vec.go]: Should we use `len([]rune(str))` here?

Comment (zhongshanhao) [reply expression/builtin_string_vec.go]: OK!Thanks for your advice.

Comment (zhongshanhao) [reply expression/builtin_string_vec.go]: > Should we use `len([]rune(str))` here?

I've taken your advice. Is that right?

Comment (zhongshanhao) [reply expression/builtin_string_vec.go]: > Should we use `len([]rune(str))` here?

I've taken your advice. Is that right?

**Pull request merged**